### PR TITLE
Update Safari data for visibilitychange event

### LINKED ISSUE #17854: api.Document.visibilitychange_event - re-update Safari & Safari on iOS compatibility
### What type of issue is this?

Missing compatibility data

### What information was incorrect, unhelpful, or incomplete?

Compatibility for `visibilitychange` event was resolved for Safari version `14.1` & Safari on iOS version `14.5` which was previously updated #10897

During a rewrite of `api/Document.json` for visibilitychange to conform to the new events structure. #15084 compatibility was downgraded back to partial compatibility. 

With the September 12, 2022 release of Safari 16, might as well update the compatibility table to be up to date with the current version 😄 

### What browsers does this problem apply to, if applicable?

_No response_

### What did you expect to see?

Oh wasn't expecting to see anything specific, just traveled down the rabbit hole.

### Did you test this? If so, how?

Yes I did test compatibility with https://event-logger.glitch.me on both mobile & desktop devices running current versions of safari. And triggering `visibilitychange` by navigating to a new page in the same window, minimizing the browser and navigating tabs. All show the `visibilitychange` event firing

### Can you link to any release notes, bugs, pull requests, or MDN pages related to this?

- Original Full Support https://github.com/mdn/browser-compat-data/pull/10897
- Document Update back to partial support https://github.com/mdn/browser-compat-data/pull/15084
- Preview Release Notes https://developer.apple.com/safari/technology-preview/release-notes/#r115
- Resolved Bug https://bugs.webkit.org/show_bug.cgi?id=151234
- Where the updates need to happen 😄 https://github.com/mdn/browser-compat-data/blob/main/api/Document.json#L6851-#L6874

### Do you have anything more you want to share?

Recommendation for updating https://github.com/mdn/browser-compat-data/blob/main/api/Document.json#L6851-#L6875 

```
"safari": [
  {
    "version_added": "16"
  },
  {
    "version_removed": "14.1",
    "version_added": "14",
    "partial_implementation": true,
    "notes": "Doesn't fire the <code>visibilitychange</code> event when navigating away from a document, so also include code to check for the <code>pagehide</code> event (which does fire for that case in all current browsers). See WebKit bugs <a href='https://webkit.org/b/116769'>116769</a>, <a href='https://webkit.org/b/151234'>151234</a>, <a href='https://webkit.org/b/151610'>151610</a>, and <a href='https://webkit.org/b/194897'>194897</a>."
  },
  {
    "version_added": "10.1",
    "partial_implementation": true,
    "notes": [
      "Doesn't fire the <code>visibilitychange</code> event when navigating away from a document, so also include code to check for the <code>pagehide</code> event (which does fire for that case in all current browsers). See WebKit bugs <a href='https://webkit.org/b/116769'>116769</a>, <a href='https://webkit.org/b/151234'>151234</a>, <a href='https://webkit.org/b/151610'>151610</a>, and <a href='https://webkit.org/b/194897'>194897</a>.",
      "Before Safari 14, the event does not bubble, so <code>document.addEventListener('visibilitychange', ...)</code> works, but <code>window.addEventListener('visibilitychange', ...)</code> does not."
    ]
  },
  {
    "version_added": "7",
    "partial_implementation": true,
    "notes": [
      "Doesn't fire the <code>visibilitychange</code> event when navigating away from a document, so also include code to check for the <code>pagehide</code> event (which does fire for that case in all current browsers). See WebKit bugs <a href='https://webkit.org/b/116769'>116769</a>, <a href='https://webkit.org/b/151234'>151234</a>, <a href='https://webkit.org/b/151610'>151610</a>, and <a href='https://webkit.org/b/194897'>194897</a>.",
      "Before Safari 14, the event does not bubble, so <code>document.addEventListener('visibilitychange', ...)</code> works, but <code>window.addEventListener('visibilitychange', ...)</code> does not.",
      "The <code>onvisibilitychange</code> event handler property is not supported."
    ]
  }
],
"safari_ios": [
  {
    "version_added": "16"
  },
  {
    "version_removed": "14.5",
    "version_added": "7",
    "partial_implementation": true,
    "notes": [
      "Doesn't fire the <code>visibilitychange</code> event when navigating away from a document, so also include code to check for the <code>pagehide</code> event (which does fire for that case in all current browsers). See WebKit bugs <a href='https://webkit.org/b/116769'>116769</a>, <a href='https://webkit.org/b/151234'>151234</a>, <a href='https://webkit.org/b/151610'>151610</a>, and <a href='https://webkit.org/b/194897'>194897</a>.",
      "Before Safari 14, the event does not bubble, so <code>document.addEventListener('visibilitychange', ...)</code> works, but <code>window.addEventListener('visibilitychange', ...)</code> does not.",
      "The <code>onvisibilitychange</code> event handler property is not supported."
    ]
  }
]
```

### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilitychange_event

### MDN metadata

<details>
<summary>MDN page report details</summary>

* Query: `api.Document.visibilitychange_event`
* Report started: 2022-09-21T21:35:02.425Z

</details>

### PR DESCRIPTION
**Pull request by admbradford**
#### Summary

Updates `visibilitychange` compatibility for Safari & Safari iOS to reflect full compatibility.
and adds Safari version 16 to compatibility list

#### Related issues

Fixes #17854

### CODE REVIEW

**Review by Boogie55 — COMMENTED** [review]
I'm confused

File: api/Document.json [hunk 1]
```diff
@@ -6870,9 +6868,29 @@
...
+                "version_added": "7",
+                "partial_implementation": true,
+                "notes": [
+                  "Doesn't fire the <code>visibilitychange</code> event when navigating away from a document, so also include code to check for the <code>pagehide</code> event (which does fire for that case in all current browsers). See WebKit bugs <a href='https://webkit.org/b/116769'>116769</a>, <a href='https://webkit.org/b/151234'>151234</a>, <a href='https://webkit.org/b/151610'>151610</a>, and <a href='https://webkit.org/b/194897'>194897</a>.",
+                  "Before Safari 14, the event does not bubble, so <code>document.addEventListener('visibilitychange', ...)</code> works, but <code>window.addEventListener('visibilitychange', ...)</code> does not.",
+                  "The <code>onvisibilitychange</code> event handler property is not supported."
+                ]
               }
             ],
-            "safari_ios": "mirror",
```
Comment (foolip) [inline api/Document.json:6875]: Are there differences between macOS and iOS, or could this be left as just "mirror"?

**Comment by foolip** [comment]
It looks like what's changed here is that https://webkit.org/b/151234 was fixed:
https://github.com/WebKit/WebKit/commit/1a31d8a37849879e6b6934b70885a823faa63ef3

That was in [WebKit trunk version 611.1.1](https://github.com/WebKit/WebKit/blob/1a31d8a37849879e6b6934b70885a823faa63ef3/Source/WebCore/Configurations/Version.xcconfig), which maps to Safari 14.1.

So rather than marking this fully supported from Safari 16, I think it should be 14.1.

There are some notes I think need updating, but the minimal edit here would be to just add a new entry in the array with `{ "version_added": "14.1" }`, and leaving everything else along.

**Comment by admbradford** [comment]
@foolip thank you for your feedback! I've updated the PR to only include the addition of `14.1`

**Comment by foolip** [comment]
Thanks @admbradford!

**Pull request merged**